pkg/controller/capacity/capacity_controller.go

```diff
@@ -289,14 +311,18 @@ func (c *Controller) capacityTargetSyncHandler(key string) error {
 	return clusterErrors.Flatten()
 }
 
-func (c *Controller) enqueueCapacityTarget(obj interface{}) {
+func (c *Controller) enqueueCapacityTargetAfter(obj interface{}, duration time.Duration) {
 	key, err := cache.MetaNamespaceKeyFunc(obj)
 	if err != nil {
 		runtime.HandleError(err)
 		return
 	}
 
-	c.capacityTargetWorkqueue.Add(key)
+	if duration > 0 {
+		c.capacityTargetWorkqueue.AddAfter(key, duration)
+	} else {
+		c.capacityTargetWorkqueue.Add(key)
+	}
```

Comment:
Is this check required in here? I'd think that `Add(key)` and `AddAfter(key, 0)` to be equivalent, and any optimization like this to be performed at the queue implementation level.